functions/summarise_data.R

```diff
@@ -0,0 +1,34 @@
+#########################################################################
+# Name of file - summarise_data.R
+# Data release - End of Life Publication
+# Original Authors - Alice Byers
+# Orginal Date - September 2019
+#
+# Written/run on - RStudio Server
+# Version of R - 3.6.1
+#
+# Description - Function to aggregate data and calculate QoM measure.
+#
+# Approximate run time - xx minutes
+#########################################################################
+
+
+summarise_data <- function(data, split_by1, split_by2, trend = FALSE){
+  
+  if(trend == FALSE){data %<>% filter(fy == max(fy))}
+  
+  data %>%
+    
+    group_by(fy, {{split_by1}}, {{split_by2}}) %>%
+    
+    summarise(deaths = sum(deaths),
+              qom    = 100 - (((sum(los) / deaths) / 182.5) * 100),
+              comm   = 182.5 * (qom / 100)) %>%
+    
+    mutate(qom  = round_half_up(qom, 1),
+           comm = round_half_up(comm, 0))
+  
+}
+
```

Comment:
There is a way to improve the function. Instead of using split_by1 and split_by2, we can use "..." to include as many columns as we want:

summarise_data <- function(data, ..., trend = FALSE){
  
  if(trend == FALSE){data %<>% filter(fy == max(fy))}
  
  data %>%
    
    group_by(fy, ...) %>%
    
    summarise(deaths = sum(deaths),
              qom    = 100 - (((sum(los) / deaths) / 182.5) * 100),
              comm   = 182.5 * (qom / 100)) %>%
    
    mutate(qom  = round_half_up(qom, 1),
           comm = round_half_up(comm, 0))
  
}

test <- summarise_data(basefile, age_grp, sex, trend = FALSE)

Reply:
Thanks Tina - that's good to know 👍 

I'm thinking, based on some of your other comments, that I'd be better dropping this function and just having a function to calculate QoM for either hospital or community. I understand the reasoning for your suggestion to structure the basefile slightly differently but think it might be more confusing to have a dataset that you can't aggregate. I'm going to try implementing the above and hopefully this will tidy up the markdown code.

Reply:
Hi Alice, I think the current basefile is fine. But I wonder if we should also create a MTD for the plots and excel workbooks?

I can give you a call tomorrow to discuss?

Reply:
Thanks Tina - I know it's not your TPP week so happy to wait until Wednesday or next week 😄

Reply:
No problem! I can give you a call on Wednesday then.